pkg/cfg/api.go

```diff
@@ -67,6 +67,26 @@ func DefaultAPIConfig() API {
 			QueryTimeoutMs:    50,
 			Prefix:            "capi",
 		},
+		// This is an intentionally large number as an intermediate refactored state.
+		// This effectively turns off the queue size limitation.
+		QueueSize: 1000000,
+		// This is left large to make the limit light.
+		MaxConcurrentUpstreamRequests: 20000,
+		// In most cases a single worker should suffice.
+		// The default is set to 4 as a precaution against bottlenecks.
+		ProcWorkers:  4,
+		LargeReqSize: 10000,
+
+		UpstreamSubRenderNumHistParams: HistogramConfig{
+			Start:      1,
+			BucketsNum: 12,
+			BucketSize: 3,
```

Comment:
I think this histogram config can be tuned. Because subRender count is either 1 or > API.resolveGlobs.

Reply:
I see your point, but I'd rather not complicate that. Also, I don't think it will play nice with the exponential buckets.

However, I can tell you for sure that we have to refactor the histogram configs in general. For example, we use the linear config for exponential histograms) So, that's definitely on the table.